orttraining/orttraining/test/gradient/gradient_ops_test.cc

```diff
@@ -203,27 +214,28 @@ void TestBroadcastableBinaryOpGrad(const std::string& op_type,
     TensorInfo B_info{{4, 1, 1, 3}, true, transformer, DataTypeImpl::GetTensorType<float>(), {"batch", "1", "1", "seq"}};
     TensorInfo Y_info{{4, 2, 3, 3}};
 
-    gradient_checker.ComputeGradientError(op_def, {A_info, B_info}, {Y_info}, &max_error);
+    gradient_checker.ComputeGradientError(op_def, {A_info, B_info}, {Y_info}, &max_error,
+                                          attributes, true, check_not_have_shape_inferencing);
     EXPECT_IS_TINY(max_error);
   }
 }
 
 TEST(GradientCheckerTest, AddGrad) {
-  TestBroadcastableBinaryOpGrad("Add");
+  TestBroadcastableBinaryOpGrad("Add", nullptr, true);
```

Comment:
>TestBroadcastableBinaryOpGrad [](start = 2, length = 29)

test cases for shape missing ?

Reply:
actually, we need shape missing UTs for all updated gradient ops

---
In reply to: [459741656](https://github.com/microsoft/onnxruntime/pull/4586#discussion_r459741656) [](ancestors = 459741656)

Reply:
the UTs are already enabled: by setting the 'check_not_have_shape_inferencing' flag of gradient checker to true. This repeats the tests by calling the AddShapeToTensorData of OpTester with 'false' and 'true'. This effectively runs tests for each op with and without shape inferencing.

Reply:
ah... I see it now.